discover/discover.ml

```diff
@@ -355,14 +387,29 @@ let features =
     "SPLICE", L[ fd_int; I "fcntl.h"; S"splice"; ];
     "TEE", L[ fd_int; I "fcntl.h"; S"tee"; ];
     "VMSPLICE", L[ fd_int; I "fcntl.h"; S"vmsplice"; ];
-    "SOCKOPT", L[
-      fd_int;
-      I "sys/socket.h"; I "netinet/in.h"; I"netinet/tcp.h";
-      S"setsockopt"; S"getsockopt";
+    "SOCKOPT", ANY[
+      [
+        fd_int;
+        I "sys/socket.h"; I "netinet/in.h"; I"netinet/tcp.h";
+        S"setsockopt"; S"getsockopt";
+      ];
+      [
+        I "winsock2.h"; I "ws2tcpip.h";
+        S"setsockopt"; S"getsockopt";
+      ]
+    ];
+    "TCP_KEEPCNT", ANY[
+      [ I "netinet/in.h"; I "netinet/tcp.h"; V "TCP_KEEPCNT" ];
+      [ I "winsock2.h"; I "ws2tcpip.h"; SVC ("TCP_KEEPCNT", "0x10", "!defined(TCP_KEEPCNT) && defined(__MINGW32__)") ];
+    ];
+    "TCP_KEEPIDLE", ANY[
+      [ I "netinet/in.h"; I "netinet/tcp.h"; V "TCP_KEEPIDLE" ];
+      [ I "winsock2.h"; I "ws2tcpip.h"; SVC ("TCP_KEEPIDLE", "0x03", "!defined(TCP_KEEPIDLE) && defined(__MINGW32__)") ];
+    ];
+    "TCP_KEEPINTVL", ANY[
+      [ I "netinet/in.h"; I "netinet/tcp.h"; V "TCP_KEEPINTVL" ];
+      [ I "winsock2.h"; I "ws2tcpip.h"; SVC ("TCP_KEEPINTVL", "0x11", "!defined(TCP_KEEPINTVL) && defined(__MINGW32__)") ];
```

Comment:
do you envision more usage for SVC? I understand currently such cases are handled just with ifdef directly in C file